lib/3scale/backend/rack/exception_catcher.rb

```diff
@@ -42,6 +40,11 @@ def call(env)
         rescue Backend::Error => e
           delete_sinatra_error! env
           respond_with e.http_code, prepare_body(e.to_xml, env)
+        rescue ArgumentError => e
+          return unhandled_exception(e) unless e.message.include?(INVALID_BYTE_SEQUENCE_ERR_MSG)
+
+          delete_sinatra_error! env
+          respond_with 400, Backend::NotValidData.new.to_xml
```

Comment:
Check https://github.com/3scale/apisonator/pull/411/commits/25d53c9feee8cb132156626d349dafc322ac0b16

Basically, an Invalid UTF8 error can be raised at any place in the code and we centralize the handling here.